src/rest.cpp

```diff
@@ -490,21 +490,22 @@ static bool rest_getutxos(HTTPRequest* req, const std::string& strURIPart)
     std::vector<bool> hits;
     bitmap.resize((vOutPoints.size() + 7) / 8);
     {
-        auto process_utxos = [&vOutPoints, &outs, &hits](const CCoinsView& view, const CTxMemPool& mempool) {
+        auto process_utxos = [&vOutPoints, &outs, &hits](const CCoinsView& view, const CTxMemPool& tx_pool) {
             for (const COutPoint& vOutPoint : vOutPoints) {
                 Coin coin;
-                bool hit = !mempool.isSpent(vOutPoint) && view.GetCoin(vOutPoint, coin);
+                bool hit = !tx_pool.isSpent(vOutPoint) && view.GetCoin(vOutPoint, coin);
                 hits.push_back(hit);
                 if (hit) outs.emplace_back(std::move(coin));
             }
         };
 
         if (fCheckMemPool) {
             // use db+mempool as cache backend in case user likes to query mempool
-            LOCK2(cs_main, mempool.cs);
+            const auto& tx_pool = EnsureMemPool();
```

Comment:
What's the rationale for using `auto` here? I generally agree with Matt that we should avoid use of `auto` unless there are clear benefits: https://github.com/bitcoin/bitcoin/pull/12120/files.

Reply:
Removed `auto`